### PR TITLE
added unit-test for functions in buildfiles.py

### PR DESCRIPTION
**Pull request by MatthiasWiesmann**
Added unit-test for functions in buildfiles.py and cleaned up some of the covered functions.

### CODE REVIEW

File: software/util/buildfiles.py [hunk 1]
```diff
@@ -32,100 +32,115 @@
...
-                    "docs/howwework.html",
-                    "docs/releases.html",
-                    "docs/faq.html",
-                    "docs/datamodel.html",
-                    "docs/developers.html",
-                    "docs/extension.html",
-                    "docs/meddocs.html",
-                    "docs/hotels.html"]
+    STATICPAGES = [
+        "docs/schemas.html",
```
Comment (pascalfleury) [inline software/util/buildfiles.py:131]: sort them?

File: software/util/buildfiles.py [hunk 2]
```diff
@@ -141,33 +156,39 @@ def sitemap(page):
     output.append("</urlset>\n")
     return "".join(output)
 
-def prtocolswap(content,protocol,altprotocol):
-    ret = content.replace("%s://schema.org" % protocol,"%s://schema.org" % altprotocol)
-    for ext in ["attic","auto","bib","health-lifesci","meta","pending"]:
-        ret = ret.replace("%s://%s.schema.org" % (protocol,ext),"%s://%s.schema.org" % (altprotocol,ext))
+
+def protocolSwap(content, protocol, altprotocol):
+    ret = content.replace("%s://schema.org" % protocol, "%s://schema.org" % altprotocol)
```
Comment (pascalfleury) [inline software/util/buildfiles.py:161]: urllib would be a candidate for this.

File: software/util/buildfiles.py [hunk 3]
```diff
@@ -181,40 +202,48 @@ def exportrdf(exportType):
...
             ?s ?p ?o;
                 schema:isPartOf <%s://attic.schema.org>.
-        }""" % (protocol,protocol)
+        }""" % (protocol, protocol)
         currentGraph.update(delattic)
 
-    formats =  ["json-ld", "turtle", "nt", "nquads", "rdf"]
-    extype = exportType[len("RDFExport."):]
+    formats = ["json-ld", "turtle", "nt", "nquads", "rdf"]
+    extype = exportType[len("RDFExport.") :]
```
Comment (pascalfleury) [inline software/util/buildfiles.py:224]: nit: in the [] I would not add space though.

**Review by pascalfleury — APPROVED** [review]

File: software/tests/test_builfiles.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+if not os.getcwd() in sys.path:
+    sys.path.insert(1, os.getcwd())
+
+import software
+import software.util.buildfiles as buildfiles
+
+
+class TestBuildFiles(unittest.TestCase):
+
+    def testProtocolSwapEmpty(self):
```
[the remark this answers is not in the document]
Comment (MatthiasWiesmann) [reply software/tests/test_builfiles.py:17]: I added a header doc, it checks functionality in the relevant module/library.

File: software/tests/test_builfiles.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            '')
+        self.assertEqual(buildfiles.protocolSwap(
+            content='Defined in http://schema.org and http://bib.schema.org',
+            protocol='http', altprotocol='ftp'),
+            'Defined in ftp://schema.org and ftp://bib.schema.org')
+
+    def testProtocols(self):
+        self.assertEqual(sorted(buildfiles.protocols()), ['http', 'https'])
+
+    def testArrayToStr(self):
```
[the remark this answers is not in the document]
Comment (MatthiasWiesmann) [reply software/tests/test_builfiles.py:29]: I think it's the same, but I'm not 100% sure and I have seen clones around various files.
So I first want to add some tests, and then see if I can get rid of it.

File: software/util/buildfiles.py (hunk 1 above)
Comment (MatthiasWiesmann) [reply software/util/buildfiles.py:131]: Done.

File: software/util/buildfiles.py (hunk 2 above)
Comment (MatthiasWiesmann) [reply software/util/buildfiles.py:161]: Maybe. I want to add more unit-tests before touching that code.

File: software/util/buildfiles.py (hunk 3 above)
Comment (MatthiasWiesmann) [reply software/util/buildfiles.py:224]: I'll run ruff on the file, it should fix the formatting.

**Pull request merged**